mk/flags.mk

```diff
@@ -35,6 +35,11 @@ YYFLAGS  += -Wno-other # --debug --verbose
 LLFLAGS  +=
 LDFLAGS += $(MARCH)
 
+
+# do not cause the "free nonheap object"-waning to cause an error
+# code generated by bison may cause gcc11 to errornously detect such a condition. 
+CXXFLAGS += -Wno-error=free-nonheap-object
```

Comment:
I would preffer previous implementation i mentioned

Reply:
That would be also be perfectly fine.

I was still looking for an article I once read regarding the use of `-Werror`. IIRC the digest was:

* use during debugging and development builds to face potential bugs
* do not use on build scripts for others, since they may use other compilers

But dropping the `-Werror` would also help, but may bring other complications.

Reply:
I think that clang compiler was complaining so maybe we need enhanced your PR with IF GCC THEN ...
Currently it will pass our builds as they are using older gcc. But clang should fail so you can try to merge current develop into your PR and we can see.Then we can do the if gcc condition.

Reply:
So maybe 
```Makefile
ifeq "$(CXX)" "g++"
CXXFLAGS += -Wno-free-nonheap-object
endif
```

Reply:
@d-otte if you agree with my last proposal (conditional usage), could you implement and test if it will work for you and then update PR? I think after this change it will be ready for merge.